hashintel/hash
# Add `nprogress` to HASH frontend and HASH.dev

## 🌟 What is the purpose of this PR?

Improve UX when navigating between pages



## 🔗 Related links





- [`nprogress` on npm](https://www.npmjs.com/package/nprogress)
- [Next.js example](https://github.com/vercel/next.js/tree/canary/examples/with-loading)
- [Asana task](https://app.asana.com/0/1203271043333970/1203388867587789/f) _(internal)_
- [Slack thread](https://hashintel.slack.com/archives/C02LG39FJAU/p1668708108226309) _(internal)_
- https://github.com/blockprotocol/blockprotocol/pull/770


## 🔍 What does this change?

HASH frontend and hash.dev show a progress bar when navigating between pages (including shallow routing). Unlike in https://github.com/blockprotocol/blockprotocol/pull/770, I've used [nextjs-progressbar](https://www.npmjs.com/package/nextjs-progressbar) which leaves smaller LOC footprint. It also allows to easily refer to `theme` `palette`.

## 📜 Does this require a change to the docs?

No

## ⚠️ Known issues

- [hash.ai](https://hash.ai) still lacks `nprogress` because its source is not in this repo.
- Playwright tests are failing but this is unrelated (other PRs have the same issue too).

## 🛡 What tests cover this?


- Manual

## ❓ How to test this?

1.  Check preview deployment
1.  Check local version (maybe)

## 📹 Demo


https://user-images.githubusercontent.com/608862/203082825-8b76c291-fc67-4ace-93dd-d116d4886dd8.mov

https://user-images.githubusercontent.com/608862/203097932-6431b009-61c9-42a3-98f2-393651a1c052.mov
